src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,43 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  selectedTodo: Todo;
+  onCloseModal: (value: null) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, onCloseModal }) => {
+  const [loaded, setLoaded] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const { userId, id, title, completed } = selectedTodo;
+
+  useEffect(() => {
+    const loadUser = async () => {
+      const loadedUser = getUser(userId);
```

Comment:
The function `getUser(userId)` should be awaited directly here to ensure the user data is properly fetched before proceeding. Use `await getUser(userId);` instead.